src/ray/gcs/gcs_actor.cc

```diff
@@ -90,6 +94,39 @@ const rpc::ActorTableData &GcsActor::GetActorTableData() const {
 rpc::ActorTableData *GcsActor::GetMutableActorTableData() { return &actor_table_data_; }
 
 void GcsActor::WriteActorExportEvent() const {
+  // If ray event is enabled and recorder present, emit actor events to the aggregator.
+  if (RayConfig::instance().enable_ray_event()) {
+    std::vector<std::unique_ptr<observability::RayEventInterface>> events;
+    switch (actor_table_data_.state()) {
+    case rpc::ActorTableData::DEPENDENCIES_UNREADY:
+      // Emit definition event only when the actor is first registered
+      // (DEPENDENCIES_UNREADY)
+      events.push_back(std::make_unique<observability::RayActorDefinitionEvent>(
+          actor_table_data_, session_name_));
```

Comment:
I'd make it explicit by passing in a bool arg like `export_definition_event` instead of relying on checking whether the state is DEPENDENCIES_UNREADY.